repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingConfigListener.scala

```diff
@@ -0,0 +1,115 @@
+ */
+package org.openrepose.core.services.httplogging
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.inject.Named
+import org.jtwig.environment.{EnvironmentConfiguration, EnvironmentConfigurationBuilder}
+import org.jtwig.exceptions.JtwigException
+import org.jtwig.value.Undefined
+import org.jtwig.value.convert.string.{DefaultStringConverter, StringConverter}
+import org.jtwig.{JtwigModel, JtwigTemplate}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.services.httplogging.config.{Format, HttpLoggingConfig, Message}
+import org.openrepose.core.services.httplogging.jtwig.HttpLoggingEnvironmentConfiguration
+import org.slf4j.LoggerFactory
+import play.api.libs.json.Json
+
+import scala.collection.JavaConverters._
+import scala.util.{Success, Try}
+
+/**
+  * Handles configuration for the [[HttpLoggingService]].
+  */
+@Named
+class HttpLoggingConfigListener extends UpdateListener[HttpLoggingConfig] with StrictLogging {
+
+  import HttpLoggingConfigListener._
+
+  private var initialized: Boolean = false
+  private var templates: List[LoggableTemplate] = List.empty
+
+  override def configurationUpdated(configurationObject: HttpLoggingConfig): Unit = {
+    logger.trace("Updating HTTP Logging Service configuration")
+
+    templates = configurationObject.getMessage.asScala
+      .toList
+      .map { message =>
+        val envConf = HttpLoggingEnvironmentConfiguration(message.getFormat)
+
+        validateFormat(message, envConf).recover {
+          case e: JtwigException =>
+            // This type of Exception is known to be caused by our test data not matching the format of real data
+            logger.warn("Unable to validate {} for {}", message.getFormat, message, e)
+          case e: JsonProcessingException =>
+            logger.warn("Failed to validate {} for {}", message.getFormat, message, e)
+        }
+
+        val template = JtwigTemplate.inlineTemplate(message.getValue, envConf)
+        LoggableTemplate(template, LoggerFactory.getLogger(message.getLogTo))
+      }
+
+    initialized = true
+
+    logger.trace("Updated HTTP Logging Service configuration")
+  }
+
+  override def isInitialized: Boolean = {
+    initialized
+  }
+
+  def loggableTemplates: List[LoggableTemplate] = {
+    templates
+  }
+}
+
+object HttpLoggingConfigListener extends StrictLogging {
+  private final val JsonValidationStringConverter: StringConverter = new DefaultStringConverter {
+    override def convert(input: Any): String = input match {
+      case Undefined.UNDEFINED => "123"
+      case _ => super.convert(input)
+    }
+  }
+
+  private final def jsonValidationEnvConf(envConf: EnvironmentConfiguration): EnvironmentConfiguration = {
+    // @formatter:off
+    new EnvironmentConfigurationBuilder(envConf)
+        .value()
+          .withStringConverter(JsonValidationStringConverter)
+        .and()
```

Comment:
It is correct. The configuration object itself is hierarchical; traversal of that hierarchy is achieve by dropping into the relevant configuration (e.g., `value`) then returning to the parent configuration (via `and`) so that you can then drop into another configuration. That's just how the API is written, so I went with it.